echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryEventListener.java

```diff
@@ -0,0 +1,162 @@
+
+  private final TelemetryConfig.TelemetryConfigProps telemetryConfigProps;
+
+  @Autowired
+  public TelemetryEventListener(
+      TelemetryService telemetryService,
+      TelemetryConfig.TelemetryConfigProps telemetryConfigProps) {
+    this.telemetryService = telemetryService;
+    this.telemetryConfigProps = telemetryConfigProps;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public void processEvent(Event event) {
+    try {
+      if (event.getDetails() == null || event.getContent() == null) {
+        log.debug("Telemetry not sent: Details or content not found in event");
+        return;
+      }
+
+      String eventType = event.getDetails().getType();
+      if (!LOGGABLE_DETAIL_TYPES.contains(eventType)) {
+        log.debug("Telemetry not sent: type '{}' not whitelisted ", eventType);
+        return;
+      }
+
+      Map<String, Object> execution = (Map<String, Object>) event.content.get("execution");
+      if (execution == null) {
+        log.debug("Missing execution from Event content.");
+        return;
+      }
+
+      String hashedApplicationId = hash(event.details.getApplication());
+      String hashedExecutionId = hash(execution.getOrDefault("id", "").toString());
+      Execution.Type executionType =
+          Execution.Type.valueOf(
+              // TODO(ttomsu, louisjimenez): Add MPTv1 and v2 execution type detection.
+              execution.getOrDefault("type", "").toString().toUpperCase());
+      Status executionStatus =
+          Status.valueOf(execution.getOrDefault("status", "").toString().toUpperCase());
+
+      Map trigger = (Map) execution.getOrDefault("trigger", new HashMap());
+      Execution.Trigger.Type triggerType =
+          Execution.Trigger.Type.valueOf(
+              trigger.getOrDefault("type", "UNKNOWN").toString().toUpperCase());
+
+      List<Map> stages = (List<Map>) execution.getOrDefault("stages", new ArrayList<>());
+      List<Stage> protoStages = stages.stream().map(this::toStage).collect(Collectors.toList());
+
+      Execution executionProto =
+          Execution.newBuilder()
+              .setId(hashedExecutionId)
+              .setType(executionType)
+              .setStatus(executionStatus)
+              .setTrigger(Execution.Trigger.newBuilder().setType(triggerType))
+              .addAllStages(protoStages)
+              .build();
+
+      Application application =
+          Application.newBuilder().setId(hashedApplicationId).setExecution(executionProto).build();
+
+      SpinnakerInstance spinnakerInstance =
+          SpinnakerInstance.newBuilder()
+              .setId(telemetryConfigProps.getInstanceId())
+              .setVersion(telemetryConfigProps.getSpinnakerVersion())
+              .setApplication(application)
+              .build();
+
+      com.netflix.spinnaker.kork.proto.stats.Event loggedEvent =
+          com.netflix.spinnaker.kork.proto.stats.Event.newBuilder()
+              .setSpinnakerInstance(spinnakerInstance)
```

Comment:
An execution should always be in the context of an application - pipelines and ad hoc tasks run _in_ applications. Basically it mirrors the proto mirrors the hierarchy of the system - a spinnaker instance --> an application under management --> a deployment (or other pipeline run) of that application

Reply:
But that's backwards, right? If you're modeling a Java object of an event, you would have an application field on the event. I can tell that's how you'd model it because that's exactly how Event is already modeled. I can't see why the proto should be different?

Reply:
Flattened the proto - here's a sample textproto:

```
spinnaker_instance: <
  id: "146a74d1-19d4-4e33-8bbc-a209b3090d74"
  version: "branch:upstream/master"
>
application: <
  id: "5be1ecc7935f1dd85635d4feedaf660594030253cc97c9e9ca3819ffeac36b65"
>
execution: <
  id: "6fc5eec30babcba3def193ba92e5c6e83b49942c72dd142032dfc33e516757f3"
  type: ORCHESTRATION
  trigger: <
    type: MANUAL
  >
  stages: <
    type: "scaleManifest"
    status: SUCCEEDED
  >
  status: SUCCEEDED
>
```

Reply:
https://github.com/spinnaker/kork/pull/381 for the proto change